website/versioned_docs/version-0.13.1/indexing.md

```diff
@@ -20,34 +24,106 @@ _Figure: Comparison of merge cost for updates (yellow blocks) against base files
+
+
+
+## Advantages and Disadvantages
+
+|                Type                |                                                                                   Advantages                                                                                   |                        Disadvantages                         |             since              |
+|:----------------------------------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:| :----------------------------------------------------------: | :----------------------------: |
+|         **Bucket(Simple)**         | 1. No additional storage of index information is required, only calculation is required<br /> 2. Can be executed on different engines<br />3. No small files will be generated |      1. Unable to scale down the number of buckets<br />       |              0.11              |
+| **Bucket<br />(Consistent Hashing)** |                                           1. Same as the bucket above<br />2. Ability to automatically adjust the number of buckets                                            |                1. Not support bulk Insert yet                | 1. Spark 0.13<br />2. Flink 0.14 |
+|          **Flink State**           |                                                                        1. Support global deduplication                                                                         | 1. Flink only<br />2. State size expands infinitely<br />3. Bootstrap takes a very long time |              0.7               |
+|             **Bloom**              |                                                                                   1. Simple                                                                                    | 1. There is a problem of false positives, and the effect is not ideal under extremely large amounts of data.<br />2. Spark Only |              0.6               |
+|             **HBase**              |                                                                        1. Support global deduplication                                                                         |             1. Depends on HBase<br />2. Spark Only             |              0.6               |
+|             **Simple**             |                                                                                   1. Simple                                                                                    |           1. Only for testing, not for production            |                                |
```

Comment:
simple index can be used for production and it's a fit for random updates spread out a lot of partitions